eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -0,0 +1,247 @@
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    protected function configure()
+    {
+        $config = $this->repositoryConfigurationProvider->getRepositoryConfig();
+
+        $this
+            ->setName('ezplatform:content:cleanup-versions')
+            ->setDescription('Remove unwanted content versions. It keeps published version untouched. By default, it keeps also the last archived/draft version.')
+            ->addOption(
+                'status',
+                't',
+                InputOption::VALUE_OPTIONAL,
+                    sprintf(
+                        "Select which version types should be removed: '%s', '%s', '%s'.",
+                        self::VERSION_DRAFT,
+                        self::VERSION_ARCHIVED,
+                        self::VERSION_ALL
+                    ),
+                self::VERSION_ALL
+            )
+            ->addOption(
+                'keep',
+                'k',
+                InputOption::VALUE_OPTIONAL,
+                "Sets number of the most recent versions (both drafts and archived) which won't be removed.",
+                $config['options']['default_version_archive_limit']
+            )
+            ->addOption(
+                'user',
+                'u',
+                InputOption::VALUE_OPTIONAL,
+                'eZ Platform username (with Role containing at least Content policies: remove, read, versionread)',
+                self::DEFAULT_REPOSITORY_USER
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (($keep = (int) $input->getOption('keep')) < 0) {
+            throw new InvalidArgumentException(
+                'status',
+                'Keep value can not be negative.'
+            );
+        }
+
+        $status = $input->getOption('status');
+
+        $contentIds = $this->getObjectsIds($keep, $status);
+        $contentIdsCount = count($contentIds);
+
+        if ($contentIdsCount === 0) {
+            $output->writeln('<info>There is no Contents matching given criteria.</info>');
+
+            return;
+        }
+
+        $output->writeln(sprintf(
+            '<info>Found %d Content IDs matching given criteria.</info>',
+            $contentIdsCount
+        ));
+
+        $removedVersionsCounter = 0;
+
+        foreach ($contentIds as $contentId) {
+            try {
+                $contentInfo = $this->contentService->loadContentInfo((int) $contentId);
+                $versions = $this->contentService->loadVersions($contentInfo);
+                $versionsCount = count($versions);
+
+                $output->writeln(sprintf(
+                    '<info>Content %d has %d version(s)</info>',
+                    (int) $contentId,
+                    $versionsCount
+                ), Output::VERBOSITY_VERBOSE);
+
+                $versions = array_slice(array_reverse($versions), $keep);
```

Comment:
AFAIKS `array_reverse` call could be ommited. Call `array_slice` and pass  second argument < 0;